airflow/models/baseoperator.py

```diff
@@ -634,6 +634,43 @@ def deps(self) -> Set[BaseTIDep]:
             NotPreviouslySkippedDep(),
         }
 
+    def set_xcomargs_dependencies(self) -> None:
```

Comment:
Does this function set lineage/inlets/outlets too? If not it should please.

Reply:
No, it only resolves up/down stream. I will take a look at the lineage

Reply:
Hm, I'm not sure how this should be approached. What I would is :
```python
            if isinstance(arg, XComArg):
                op: BaseOperator = arg.operator
                self.set_upstream(op)
                # Picks up any outlets from direct upstream tasks that have outlets defined,
                # as such that if A -> B and B does not have inlets but A has outlets, then
                # these are provided as inlets to B.
                self.add_inlets(op.get_outlet_defs())
```
However, I am not sure if this is the right approach. Not all outlets of A has to be inlets of B. Proposed behaviour is what we get when we use `inlets=AUTO` and I'm not sure this is something that should be done auto of the box.

Reply:
Hmmm not too familiary with lineage yet.

/cc @bolkedebruin if you have time to look at this?

Reply:
What would you say to remove lineage from the scope if this PR? I can create an issue addressing this feature.

Reply:
Yeah, sounds good to me.